tests/console/verify_efi_mok.pm

```diff
@@ -0,0 +1,275 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+# Summary: Check EFI boot in images or after OS installation
+# Maintainer: Martin Loviska <mloviska@suse.com>
+
+use Mojo::Base 'opensusebasetest';
+use testapi;
+use utils qw(zypper_call is_efi_boot);
+use version_utils qw(is_leap is_opensuse is_sle);
+use Utils::Backends qw(is_hyperv is_svirt_except_s390x);
+use jeos qw(reboot_image set_grub_gfxmode);
+use constant {
+    SYSFS_EFI_BITS      => '/sys/firmware/efi/fw_platform_size',
+    FSTAB               => '/etc/fstab',
+    GRUB_DEFAULT        => '/etc/default/grub',
+    GRUB_CFG            => '/boot/grub2/grub.cfg',
+    SYSCONFIG_BOOTLADER => '/etc/sysconfig/bootloader'
+};
+
+sub get_expected_efi_settings {
+    my $settings      = {};
+    my $efi_rec_label = (is_opensuse() ? (lc get_var('DISTRI')) : 'sles');
+    my $efi_exec      = $efi_rec_label . '/grubx64.efi';
+
+    if (!get_var('DISABLE_SECUREBOOT', 0)) {
+        $efi_exec = $efi_rec_label . '/shim.efi';
+        $efi_rec_label .= '-secureboot';
+    }
+    $settings->{exec}  = '/EFI/' . $efi_exec;
+    $settings->{label} = $efi_rec_label;
+
+    return $settings;
+}
+sub efibootmgr_current_boot {
+    my $ebm_raw = script_output 'efibootmgr --verbose';
+    my $h       = {};
+
+    ($h->{bootid}) = $ebm_raw =~ /^BootCurrent:\s+(\d+)/m;
+    die("Missing BootCurrent: in efibootmgr output") unless (exists($h->{bootid}) or defined($h->{bootid}));
```

Comment:
```suggestion
    die("Missing BootCurrent: in efibootmgr output") unless (defined($h->{bootid}));
```
as `$h->{bootid}` will exists always but undefined if it hadn't a match.

see:
perl -we "$(cat <<'EOT'
my $h = {};
my $s = "FOO BAR";
($h->{test}) = $s =~ /WILL_NOT(.*)_MATCH/m;
print "exists$/" if (exists($h->{test}));
print "defined$/" if (defined($h->{test}));
EOT
)"

Reply:
Seems like I should be concerned about `label` key only. Thanks! :)